noir-projects/aztec-nr/aztec/src/context/private_context.nr

```diff
@@ -220,6 +221,31 @@ impl PrivateContext {
         keys.app_nullifier_secret_key
     }
 
+    // TODO: (#6176) Replace request_app_nullifier_secret_key above with this once we no longer get app nullifier secret key with address
+    pub fn request_nsk_app_with_npk_m_h(&mut self, master_public_nullifying_key_hash: Field) -> Field {
+        let keys = if self.nullifier_key.is_none() {
```

Comment:
There is a caching issue here, see #6296, we will address it in a separate pr to not make this larger than it is.